test/infrastructure/inmemory/pkg/server/proxy/dial.go

```diff
@@ -33,85 +33,80 @@ import (
 
 const defaultTimeout = 10 * time.Second
 
-// Dialer creates connections using Kubernetes API Server port-forwarding.
+// Dialer opens connections via Kubernetes API server port-forward.
+//
+// Safe for concurrent use: each dial builds its own SPDY transport and http.Client
+// to avoid races.
 type Dialer struct {
-	proxy          Proxy
-	clientset      *kubernetes.Clientset
-	proxyTransport http.RoundTripper
-	upgrader       spdy.Upgrader
-	timeout        time.Duration
+	proxy     Proxy
+	clientset *kubernetes.Clientset
+	timeout   time.Duration
 }
 
-// NewDialer creates a new dialer for a given API server scope.
+// NewDialer returns a Dialer for proxy p.
+// Options may set timeouts or other fields. The timeout also updates p.KubeConfig.
 func NewDialer(p Proxy, options ...func(*Dialer) error) (*Dialer, error) {
 	if p.Port == 0 {
 		return nil, errors.New("port required")
 	}
 
-	dialer := &Dialer{
-		proxy: p,
-	}
+	d := &Dialer{proxy: p}
 
 	for _, option := range options {
-		err := option(dialer)
-		if err != nil {
+		if err := option(d); err != nil {
 			return nil, err
 		}
 	}
 
-	if dialer.timeout == 0 {
-		dialer.timeout = defaultTimeout
-	}
-	p.KubeConfig.Timeout = dialer.timeout
-	clientset, err := kubernetes.NewForConfig(p.KubeConfig)
-	if err != nil {
-		return nil, err
+	if d.timeout == 0 {
+		d.timeout = defaultTimeout
 	}
-	proxyTransport, upgrader, err := spdy.RoundTripperFor(p.KubeConfig)
+	p.KubeConfig.Timeout = d.timeout
+
+	cs, err := kubernetes.NewForConfig(p.KubeConfig)
```

Comment:
I'd like to keep the names we had previously for dialer, clientset etc, so its better comparable to controlplane/kubeadm/internal/proxy/dial.go